### PR TITLE
Added DCDATE (MAJORBBS.157)

### LINKED ISSUE #99: [CHRGLOB] Bud Globals - Crash on Initilization
**Module Information**
* Identifier: CHRGLOB
* Module Name: Bud Globals
* Version: 1.1

**Describe the bug**
Fails to initialize

**To Reproduce**
1. Add to modules.json
2. MBBSEMU fails to load module
3. Example of module where .DAT files **do not** autogenerate, had to create all 4 dat files manually from .VIR (CHRGLOB, CHRMACRO, CHRRCNT, CHRUGRAM)

**Expected behavior**
Module loads

**Screenshots**
```
Critical Exception has occurred:
System.ArgumentOutOfRangeException: Specified argument was out of the range of valid values. (Parameter 'Unknown Exported Function Ordinal in MAJORBBS: 157')
   at MBBSEmu.HostProcess.ExportedModules.Majorbbs.Invoke(UInt16 ordinal, Boolean offsetsOnly)
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.ExternalFunctionDelegate(UInt16 ordinal, UInt16 functionOrdinal)
   at MBBSEmu.CPU.CpuCore.Tick()
   at MBBSEmu.HostProcess.ExecutionUnits.ExecutionUnit.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.Module.MbbsModule.Execute(IntPtr16 entryPoint, UInt16 channelNumber, Boolean simulateCallFar, Boolean bypassSetState, Queue`1 initialStackValues, UInt16 initialStackPointer)
   at MBBSEmu.HostProcess.MbbsHost.AddModule(MbbsModule module)
   at MBBSEmu.Program.Run(String[] args)
```

**Software Information:**
- OS: Windows Server 2019 Standard
- Telnet Client: Netrunner

### PR DESCRIPTION
**Pull request by enusbaum**
- Added DCDATE() Ordinal (MAJORBBS.157)
- Unit Tests for DCDATE()
- Unit Tests for NCDATE()

Fixes #99

### CODE REVIEW

File: MBBSEmu.Tests/ExportedModules/Majorbbs/dcdate_Tests.cs [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+namespace MBBSEmu.Tests.ExportedModules.Majorbbs
+{
+    public class dcdate_Tests : ExportedModuleTestBase
+    {
+        private const int DCDATE_ORDINAL = 157;
+
+        [Theory]
+        [InlineData("09/17/20", 20785)]
+        [InlineData("12/31/90", 5535)]
+        [InlineData("1/1/80", 33)]
```
Comment (paladine) [inline MBBSEmu.Tests/ExportedModules/Majorbbs/dcdate_Tests.cs:15]: needs some tests that generate -1

File: MBBSEmu.Tests/ExportedModules/Majorbbs/ncdate_Tests.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+namespace MBBSEmu.Tests.ExportedModules.Majorbbs
+{
+    public class ncdate_Tests : ExportedModuleTestBase
+    {
+        private const int NCDATE_ORDINAL = 428;
+
+        [Theory]
+        [InlineData(20785, "09/17/20")]
+        [InlineData(5535, "12/31/90")]
+        [InlineData(33, "01/01/80")]
```
Comment (paladine) [inline MBBSEmu.Tests/ExportedModules/Majorbbs/ncdate_Tests.cs:14]: would be interesting to see what passing 0/-1 here does, as that's an invalid date

File: MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs [hunk 3]
```diff
@@ -7494,5 +7496,20 @@ private void rtstcrd()
...
+        /// <summary>
+        ///     Decode string date formatted 'MM/DD/YY' to int format YYYYYYYMMMMDDDDD
+        ///
+        ///     Signature: int date=dcdate(char *ascdat);
+        /// </summary>
+        private void dcdate()
+        {
+            var inputDate = GetParameterString(0, true);
+
+            var inputDatetime = DateTime.Parse(inputDate);
```
Comment (paladine) [inline MBBSEmu/HostProcess/ExportedModules/Majorbbs.cs:7510]: need to catch a parse exception here and have it return -1

**Review by paladine — CHANGES_REQUESTED** [review]

File: MBBSEmu.Tests/ExportedModules/Majorbbs/ncdate_Tests.cs (hunk 2 above)
Comment (enusbaum) [reply MBBSEmu.Tests/ExportedModules/Majorbbs/ncdate_Tests.cs:14]: Verified and it just returns a jacked up date -- formatted valid, but the values are just wrong.

**Review by paladine — APPROVED** [review]

**Pull request merged**